horace_core/sqw/file_io/@sqw_binfile_common/put_det_info.m

```diff
@@ -18,6 +18,8 @@
 obj=obj.check_obj_initated_properly();
 %
 [detpar,new_obj] = obj.extract_correct_subobj('detpar',argi{:});
+% converting to old detpar representation for old file format
+detpar = detpar{1}.get_detpar_representation();
 if new_obj
```

Comment:
to be facy, I do not like this name. 
detpar representation is now `detector_array`.  and If I understand this piece of code correctly, it should be `get_legacy_detpar` (the structure)

Would it be better to rename `get_detpar_representyation` to `get_detpar_legacy` ?

Reply:
This code is processing detector information for output to the old file format, which actually needs the detpar struct. I have added additional commenting to make this clearer, including that the detpar{1} is now the detector arrays. It is unclear whether writing to the old file format should need to be supported at this stage, but while it is, this process is not legacy. The function name here is a reasonable neutral description of what it does. NB format is v2 from the loader in the calling test test_sqw/test_save::test_save_with_loader